### PR TITLE
Check jinja version for consent resource

### PR DESCRIPTION
**Pull request by richvdh**
Fixes https://github.com/matrix-org/synapse/issues/3296

### CODE REVIEW

**Comment by erikjohnston** [comment]
I'm failing to follow the changes that you're making here? It seems to be doing more than just adding checking jinja versions?

File: synapse/python_dependencies.py [hunk 1]
```diff
@@ -83,19 +87,31 @@ def list_requirements():
...
 
 class DependencyException(Exception):
+    @property
+    def message(self):
+        return "\n".join([
+            "Missing Requirements: %s" % (", ".join(self.dependencies),),
+            "To install run:",
+            "    pip install --upgrade --force %s" % (" ".join(self.dependencies),),
+            "",
+        ])
```
Comment (erikjohnston) [inline synapse/python_dependencies.py]: I'd be tempted to pull this up as a top level template string and then use the multiline quotes, but that's a matter of taste

Comment (richvdh) [reply synapse/python_dependencies.py]: yeah, that would have been a good idea. I just c&ped it tbh.

Comment (erikjohnston) [reply synapse/python_dependencies.py]: Fair

**Pull request merged**